tests/storage/test_roommember.py

```diff
@@ -161,6 +161,60 @@ def test_get_joined_users_from_context(self):
         )
         self.assertEqual(users.keys(), {self.u_alice, self.u_bob})
 
+    def test_store_room_members_txn(self):
+        def _store_room_members_txn(txn, events, backfilled):
+            """Store a room member in the database."""
+
+            def non_null_str_or_none(val) -> str:
+                return val if isinstance(val, str) and "\u0000" not in val else None
+
+            self.store.db_pool.simple_insert_many_txn(
+                txn,
+                table="room_memberships",
+                values=[
+                    {
+                        "event_id": event.event_id,
+                        "user_id": "random_test_value",
+                        "sender": event.user_id,
+                        "room_id": event.room_id,
+                        "membership": event.membership,
+                        "display_name": non_null_str_or_none(
+                            event.content.get("displayname")
+                        ),
+                        "avatar_url": non_null_str_or_none(
+                            event.content.get("avatar_url")
+                        ),
+                    }
+                    for event in events
+                ],
+            )
+
+        room = self.helper.create_room_as(self.u_alice, tok=self.t_alice)
+        bob_event = self.get_success(
+            event_injection.inject_member_event(
+                self.hs, room, self.u_bob, Membership.JOIN
+            )
+        )
+
+        # Create an event with a null-containing string to pass to _store_room_members_txn
+        event, context = self.get_success(
+            event_injection.create_event(
+                self.hs,
+                room_id=room,
+                sender=self.u_alice,
+                prev_event_ids=[bob_event.event_id],
+                type="m.test.1",
+                content={"displayname": "bad\u0000null", "membership": Membership.JOIN},
+            )
+        )
+
+        # pass event to _store_room_members_txn and verify that it doesn't blow up
+        self.get_success(
+            self.store.db_pool.runInteraction(
+                "store room members", _store_room_members_txn, [event], False
```

Comment:
```suggestion
                "store room members", self.hs.get_storage().persistence.persist_events_store._store_room_members_txn, [event], False
```
That one was hard to dig up, but doing this lets you avoid copying the function (so there won't be any chance we forget to copy it into the test if we change/break it in the future)

Reply:
I don't know if you really need to test this function explicitly/individually if you properly demonstrate that `self.helper.change_membership(` (as discussed in another thread) works (assuming it didn't work before!).

Reply:
Thanks for showing how to access `_store_room_members_txn`-even though it seems like it's not necessary to call now-I really scratched my head over how to get access to it so I appreciate being shown the answer!

Reply:
Oh interesting-thanks for pointing that out.